core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransactionManager.java

```diff
@@ -68,9 +74,17 @@ public Optional<String> getTable() {
     return tableName;
   }
 
-  protected TwoPhaseCommitTransaction activate(TwoPhaseCommitTransaction transaction)
+  protected TwoPhaseCommitTransaction decorate(TwoPhaseCommitTransaction transaction)
```

Comment:
`decorate()` method decorates the transaction object with the `TwoPhaseCommitTransactionDecorator` instances.